src/main/java/picard/arrays/CreateBafRegressMetricsFile.java

```diff
@@ -0,0 +1,119 @@
+ * THE SOFTWARE.
+ */
+
+package picard.arrays;
+
+import htsjdk.samtools.metrics.MetricsFile;
+import htsjdk.samtools.util.IOUtil;
+import org.broadinstitute.barclay.argparser.Argument;
+import org.broadinstitute.barclay.argparser.CommandLineProgramProperties;
+import org.broadinstitute.barclay.help.DocumentedFeature;
+import picard.PicardException;
+import picard.cmdline.CommandLineProgram;
+import picard.cmdline.StandardOptionDefinitions;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * A simple program to create a standard picard metrics file
+ * from the output of bafRegress
+ */
+@CommandLineProgramProperties(
+        summary = CreateBafRegressMetricsFile.USAGE_DETAILS,
+        oneLineSummary = "Program to generate a picard metrics file from the output of the bafRegress tool.",
+        programGroup = picard.cmdline.programgroups.GenotypingArraysProgramGroup.class
+)
+
+@DocumentedFeature
+public class CreateBafRegressMetricsFile extends CommandLineProgram {
+    static final String USAGE_DETAILS =
+            "CreateBafRegressMetricsFile takes an output file as generated by the bafRegress tool and creates a picard metrics file. " +
+                    "BAFRegress <a href='https://genome.sph.umich.edu/wiki/BAFRegress'>bafRegress</a> " +
+                    "is a software that detects and estimates sample contamination using B allele frequency data from Illumina genotyping arrays using a regression model." +
+                    "<h4>Usage example:</h4>" +
+                    "<pre>" +
+                    "java -jar picard.jar CreateBafRegressMetricsFile \\<br />" +
+                    "      INPUT=bafRegress.output.txt \\<br />" +
+                    "      OUTPUT=outputBaseName" +
+                    "</pre>";
+
+    @Argument(shortName = StandardOptionDefinitions.INPUT_SHORT_NAME, doc = "The output of bafRegress (typically captured stdout).")
+    public File INPUT;
+
+    @Argument(shortName = StandardOptionDefinitions.OUTPUT_SHORT_NAME, doc = "Basename for the metrics file that will be written." +
+            " Resulting file will be <OUTPUT>." + FILE_EXTENSION)
+    public File OUTPUT;
+
+    public static final String FILE_EXTENSION = "bafregress_metrics";
+
+    @Override
+    protected int doWork() {
+        IOUtil.assertFileIsReadable(INPUT);
+        final File metricsFile = new File(OUTPUT + "." + FILE_EXTENSION);
+        IOUtil.assertFileIsWritable(metricsFile);
+
+        final MetricsFile<BafRegressMetrics, ?> bafRegressMetricsMetricsFile = getMetricsFile();
+
+        final Pattern HEADER_PATTERN = Pattern.compile("^sample\\s+estimate\\s+stderr\\s+tval\\s+pval\\s+callrate\\s+Nhom\\s*$");
+        try (BufferedReader br = new BufferedReader(new FileReader(INPUT))) {
+            String line;
+            line = br.readLine();
+            lineMatch(line, HEADER_PATTERN);
```

Comment:
Is a regex needed here? Is this to support both spaces and tabs? Otherwise you could do a direct string compare.